charts/cannon/templates/statefulset.yaml

```diff
@@ -33,6 +40,10 @@ spec:
             # Instead we sleep for a day. (SIGTERM is still sent, but after the preStop completes)
             exec:
               command: ["sleep", {{ .Values.drainTimeout | quote }} ]
+        env:
+        - name: CANNON_SERVICE_NAME
+          # Should match the name of the headless-service
+          value: cannon
```

Comment:
This should probably be a template variable, but pretty much everywhere else just hardcodes cannon as well and I'm not super confident in changing them all...

Reply:
Yeah, I think names should be "hardcoded"; what would be the use case to not do it? Scoping is done at the namespace level

Reply:
It can be hardcoded; but I think it's nice to somehow distinguish between all of the different 'cannon's that exist; for instance the `wireService` cannon is just a selector; which is a different value (semantically) than the name of the cannon's service, which is different than the name of the stateful set, which is different from the name of the configmap; etc.

Even from a readability standpoint it makes it easier to see why each value is set to what; this env var is specifically set to the SERVICE's name; which just so happens to be cannon.

Reply:
I went ahead and parameterized it; take a peek 😄